docs/server/source/production-deployment-template/architecture.rst

```diff
@@ -0,0 +1,69 @@
+Architecture of an IPDB Node
+============================
+
+An IPDB Production deployment is hosted on a Kubernetes cluster and includes:
+
+* NGINX, OpenResty, BigchainDB and MongoDB
+  `Kubernetes Services <https://kubernetes.io/docs/concepts/services-networking/service/>`_.
+* NGINX, OpenResty, BigchainDB, Monitoring Agent and Backup Agent
+  `Kubernetes Deployments <https://kubernetes.io/docs/concepts/workloads/controllers/deployment/>`_.
+* MongoDB `Kubernetes StatefulSet <https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/>`_.
+* External systems like `3scale <https://3scale.net>`_,
+  `MongoDB Cloud Manager <https://cloud.mongodb.com>`_ and the
+  `Azure Operations Management Suite
+  <https://docs.microsoft.com/en-us/azure/operations-management-suite/>`_.
+
+.. image:: ../_static/arch.svg
+
+
+We describe the role of each of these modules here.
+
+NGINX
+-----
+
+We use an NGINX as HTTP proxy on port 443 at the cloud entrypoint for:
+
+#. Rate Limiting: We configure NGINX to allow only a certain number of requests
+   (configurable) which prevents DoS attacks.
+
+#. HTTPS Termination: The HTTPS connection does not carry through all the way
+   to BigchainDB and terminates at NGINX for now.
+
+#. Request Routing: For HTTPS connections on port 443 (or the configured BigchainDB public api port),
+   the connection is proxied to:
+
+   #. OpenResty Service if it is a POST request.
+   #. BigchainDB Service if it is a GET request.
+
+
+We use an NGINX TCP proxy on port 27017 at the cloud entrypoint for:
```

Comment:
do we need to expose specific ports here?

Reply:
These are public ports and are always exposed.
I have added that these are configurable.